### DISCUSSION TITLE
wildcard support in deadwood

### DISCUSSION THREAD

**Discussion by zeeveisenberg** [samboy/MaraDNS#119] [Q&A] [answered]
Hi,

Is it possible to use wildcard entries in Deadwood, similar to this:
ip4["*.example.com"] = "X"
instead of having to guess all of the hostnames (evil.example.com, phish.example.com, evil098321.example.com) ?

 Or do you have to create a zonefile for every domain and use the csv2[] commands to get wildcards to work?

**Comment by samboy** [comment, marked as answer]
One can use a block hash file to get wildcard support. See https://github.com/samboy/MaraDNS/blob/master/deadwood-github/doc/Deadwood.md in the section `blocked_hosts_hash_file`:

```
Block hash files have wildcard support. For example, if "example.com" 
is in the block hash file, then deadwood will block "example.com", 
"anyname.example.com", "anything.else.example.com", 
"12345.example.com", and so on. 

Note that wildcards only work for domain names which are one, two, or 
three labels long in the database. If "really.bad.example.com" is in 
the database, "buzz.really.bad.example.com" will not match, since 
"really.bad.example.com" has four (i.e. more than three) labels. 
```

Also relevant is the man page for the tool which generates the block hash files, https://github.com/samboy/MaraDNS/blob/master/deadwood-github/doc/blockHashMake.md

(Edit: It’s time I have markdown files for the man pages, since my tools for making man pages already can generate them)

Edit 2: Example of block hash files in action are at https://github.com/samboy/BlockHash
